tsl/src/compression/compression.c

```diff
@@ -3225,15 +3241,19 @@ decompress_batches_using_index(RowDecompressor *decompressor, Relation index_rel
 			index_close(index_rel, AccessShareLock);
 			report_error(result);
 		}
-		row_decompressor_decompress_row(decompressor, NULL);
+		row_decompressor_decompress_row_to_table(decompressor);
 		heap_freetuple(compressed_tuple);
 		*chunk_status_changed = true;
 	}
-	elog(DEBUG1,
-		 "Number of compressed rows fetched from index: %d. "
-		 "Number of compressed rows filtered by orderby columns: %d.",
-		 num_segmentby_filtered_rows,
-		 num_orderby_filtered_rows);
+
+	if (ts_guc_debug_compression_path_info)
+	{
+		elog(INFO,
+			 "Number of compressed rows fetched from index: %d. "
+			 "Number of compressed rows filtered by orderby columns: %d.",
+			 num_segmentby_filtered_rows,
+			 num_orderby_filtered_rows);
+	}
 
```

Comment:
Had to change this to INFO + GUC because there are spurious DEBUG messages that make tests flaky (e.g. about new wal file being created).